### DISCUSSION TITLE
SELF HOST. DOCKER COMPOSE AND VOLUMES

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#9879] [Help] [answered]
HI all,

I am a bit confused from documentation. Regarding deploying with docker there are 2 pages:
- https://docs.budibase.com/docs/docker
- https://docs.budibase.com/docs/docker-compose

One is presenting  the installation from single image ant the other an analytical docker compose with all images budibase needs to run. 
1st question: What is the correct to use? Why there are 2 ways to install with docker?

In single image is suggested to mount data folder to an external directory that will have minio and couch db info.
On docker-compose version docker volumes are used. 
So 2nd question is whether to map data folders of budibase containers to local directories or volumes and which of those are important to backup?

**Comment by shogunpurple** [comment, marked as answer]
Hi @gmentzik

- There is no "correct" way to deploy it - it's just a matter of personal preference. If you want an all in one image and want budibase to be completely self contained including redis/database etc, the single image will suit your needs. If you need more control, have an existing couch/redis etc or just prefer to run the services as separate containers, then compose may work better. It's not a huge difference either way
- For volumes or bind mounts - that part is up to you. As with most docker setups, it's easier to backup bind mounts since they are just directories on your machine. Just back up the `/data` directory in the single image. For compose, you can use the backups CLI to backup your budibase instance.
